File: src/main/java/me/kuku/yuq/dao/ConfigDao.java
package me.kuku.yuq.dao;

import com.icecreamqaq.yudb.YuDao;
import com.icecreamqaq.yudb.jpa.annotation.Dao;
import com.icecreamqaq.yudb.jpa.annotation.Select;
import me.kuku.yuq.entity.ConfigEntity;

import java.util.List;

@Dao
public interface ConfigDao extends YuDao<ConfigEntity, Integer> {
    @Select("from ConfigEntity")
    List<ConfigEntity> findAll();
    ConfigEntity findByType(String type);
}

File: src/main/kotlin/me/kuku/yuq/dao/ConfigDao.kt
package me.kuku.yuq.dao

import com.icecreamqaq.yudb.YuDao
import com.icecreamqaq.yudb.jpa.annotation.Dao
import com.icecreamqaq.yudb.jpa.annotation.Select
import me.kuku.yuq.entity.ConfigEntity

@Dao
interface ConfigDao: YuDao<ConfigEntity, Int>{
    @Select("from ConfigEntity")
    fun findAll(): List<ConfigEntity>
    fun findByType(type: String): ConfigEntity?
}